File: src/main/java/me/aov/sellgui/utils/ColorUtils.java
package me.aov.sellgui.utils;

import net.md_5.bungee.api.ChatColor;
import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class ColorUtils {

    private static final Pattern HEX_PATTERN = Pattern.compile("&#([A-Fa-f0-9]{6})");

    public static String color(String text) {
        if (text == null) {
            return null;
        }
        Matcher matcher = HEX_PATTERN.matcher(text);
        StringBuffer buffer = new StringBuffer(text.length() + 4 * 8);
        while (matcher.find()) {
            String group = matcher.group(1);
            matcher.appendReplacement(buffer, ChatColor.of("#" + group).toString());
        }
        return org.bukkit.ChatColor.translateAlternateColorCodes('&', matcher.appendTail(buffer).toString());
    }
}

File: src/main/kotlin/me/aov/sellgui/utils/ColorUtils.kt
package me.aov.sellgui.utils

import net.md_5.bungee.api.ChatColor
import java.util.regex.Pattern

object ColorUtils {
    private val hexPattern = Pattern.compile("&#([A-Fa-f0-9]{6})")

    @JvmStatic
    fun color(text: String?): String? {
        if (text == null) return null

        val matcher = hexPattern.matcher(text)
        val buffer = StringBuffer(text.length + 4 * 8)
        while (matcher.find()) {
            matcher.appendReplacement(buffer, ChatColor.of("#${matcher.group(1)}").toString())
        }
        return org.bukkit.ChatColor.translateAlternateColorCodes('&', matcher.appendTail(buffer).toString())
    }

    @JvmStatic
    fun stripColor(text: String?): String? = text?.let { org.bukkit.ChatColor.stripColor(color(it)) }
}
